angular-ngrx-scss/src/app/state/profile/profile.selectors.ts

```diff
@@ -9,3 +9,36 @@ export const selectProfile = createSelector(
   selectProfileState,
   (state: ProfileState) => state,
 );
+
+export const selectSortAndFilter = createSelector(
+  selectProfileState,
+  (state: ProfileState) => state.sortAndFilter,
+);
+
+export const selectFilterBySearch = createSelector(
+  selectProfileState,
+  (state: ProfileState) => state.sortAndFilter?.search,
+);
+
+export const hasActiveSortAndFilters = createSelector(
+  selectFilterBySearch,
+  (search) => !!search,
+);
+
+export const selectRepos = createSelector(
+  selectProfileState,
+  selectFilterBySearch,
+  (state: ProfileState, search?: string) => {
+    if (search && search !== '') {
+      return state.repos?.filter((item) =>
+        item.name.toLowerCase().includes(search),
+      );
+    }
```

Comment:
question(requirements): Were we supposed to implement search on client-side or server-side? Client-side is fine if GitHub doesn't expose server-side search to us I suppose. I have concerns about how this would work with pagination in the event that all repos are not visible in the view. Also if server-side search is available to us, the algorithm GitHub uses will probably return more helpful results.

Reply:
With the GraphQL API, it's not possible to do search at the profile level on repos so we decided to implement this on the frontend. However, I don't know if this is possible with the HTTP API. It's a known bad UX but it's a limitation of the tools available.

Reply:
If that works, that'd be awesome but that looks like it's trying to do global search and not profile level search. Does seem like it does the profile though. I think let's leave this as is for now and write a follow up enhancement for the API version since we can do this with HTTP.